### PR TITLE
Refactoring: introduce MasterEnvironmentRunnableContext to be passed to MasterEnvironmentRunnable

### PR DESCRIPTION
**Pull request by wyzhang**

### CODE REVIEW

File: cdap-kubernetes/src/main/java/io/cdap/cdap/master/environment/k8s/KubeMasterEnvironment.java [hunk 1]
```diff
@@ -196,6 +197,12 @@ public MasterEnvironmentRunnable createRunnable(MasterEnvironmentContext context
     return cls.getConstructor(MasterEnvironmentContext.class, MasterEnvironment.class).newInstance(context, this);
   }
 
+  @Override
+  public MasterEnvironmentRunnable createRunnable(MasterEnvironmentRunnableContext context,
```
Comment (chtyim) [inline cdap-kubernetes/src/main/java/io/cdap/cdap/master/environment/k8s/KubeMasterEnvironment.java:194]: Why there are two methods for creating the Runnable? There should only be one.

Comment (wyzhang) [reply cdap-kubernetes/src/main/java/io/cdap/cdap/master/environment/k8s/KubeMasterEnvironment.java:194]: Existing createRunnable(MasterEnvironmentContext context.....) in MasterEnv is part of SPI. I suppose we cannot remove the existing one.  Marked it as deprecated 

  @Deprecated
  MasterEnvironmentRunnable createRunnable(MasterEnvironmentContext context,
                                           Class<? extends MasterEnvironmentRunnable> runnableClass) throws Exception;

Comment (chtyim) [reply cdap-kubernetes/src/main/java/io/cdap/cdap/master/environment/k8s/KubeMasterEnvironment.java:194]: We can just remove it. The SPI is very new and we should try to keep it clean for now

Comment (wyzhang) [reply cdap-kubernetes/src/main/java/io/cdap/cdap/master/environment/k8s/KubeMasterEnvironment.java:194]: Done. Removed

**Review by chtyim — APPROVED** [review]

**Pull request merged**